lib/Basics/Mutex.cpp

```diff
@@ -33,99 +33,58 @@ using namespace arangodb;
 // --SECTION                                              TRI_HAVE_POSIX_THREADS
 // -----------------------------------------------------------------------------
 
-#if defined(TRI_HAVE_POSIX_THREADS)
-
-#ifdef ARANGO_ENABLE_DEADLOCK_DETECTION
+#ifdef ARANGODB_ENABLE_MAINTAINER_MODE
 // initialize _holder to "maximum" thread id. this will work if the type of
 // _holder is numeric, but will not work if its type is more complex.
 Mutex::Mutex()
-    : _mutex(), _holder((std::numeric_limits<decltype(_holder)>::max)()) {
+  : _holder((std::numeric_limits<decltype(_holder)>::max)()) {}
 #else
-Mutex::Mutex() : _mutex() {
+Mutex::Mutex() {}
 #endif
-  pthread_mutexattr_init(&_attributes);
 
-#ifdef __linux__
-#ifdef ARANGO_ENABLE_DEADLOCK_DETECTION
-  // use an error checking mutex if available (only for LinuxThread) and only
-  // in maintainer mode
-  pthread_mutexattr_settype(&_attributes, PTHREAD_MUTEX_ERRORCHECK_NP);
-#endif
-#endif
-  
-  pthread_mutex_init(&_mutex, &_attributes);
-}
 
-Mutex::~Mutex() {
-  pthread_mutex_destroy(&_mutex);
-  pthread_mutexattr_destroy(&_attributes);
-}
+Mutex::~Mutex() {}
 
 void Mutex::lock() {
-#ifdef ARANGO_ENABLE_DEADLOCK_DETECTION
+#ifdef ARANGODB_ENABLE_MAINTAINER_MODE
   // we must not hold the lock ourselves here
   TRI_ASSERT(_holder != Thread::currentThreadId());
 #endif
 
-  int rc = pthread_mutex_lock(&_mutex);
+  _mutex.lock();
 
-  if (rc != 0) {
```

Comment:
There is a change in behavior change here:
Before, if locking the mutex produced an error, it was detected, logged, and the program was aborted.
Now, `std::mutex::lock` will throw an exception in case of an error.
I am not saying this is bad per se, but it is a change in behavior. Do you want to mimic the old behavior of logging errors in case the mutex cannot be acquired?